recipes/libcoap/all/conanfile.py

```diff
@@ -60,9 +60,14 @@ def configure(self):
 
     def source(self):
         tools.get(**self.conan_data["sources"][self.version])
```

Comment:
```suggestion
        tools.get(**self.conan_data["sources"][self.version],
                  destination=self._source_subfolder, strip_root=True)
```

You can replace all source(self) content using it. The destination + strip_root, will do all magic related to folder name.

Reply:
Sweet!